code/go/internal/validator/semantic/validate_docs_structure.go

```diff
@@ -0,0 +1,242 @@
+func ValidateDocsStructure(fsys fspath.FS) specerrors.ValidationErrors {
+	config, err := shouldValidateDocsStructure(fsys)
+	if err != nil {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredError(
+				fmt.Errorf("failed to determine if documentation structure validation should be performed: %w", err),
+				specerrors.UnassignedCode,
+			),
+		}
+	}
+	if config == nil {
+		return nil
+	}
+
+	enforcedSections, err := readDocsStructureEnforcedConfig(fsys, config)
+	if err != nil {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredError(
+				fmt.Errorf("failed to read enforced documentation structure configuration: %w", err),
+				specerrors.UnassignedCode,
+			),
+		}
+	}
+
+	err = validateReadmeStructure(fsys, enforcedSections)
+
+	if err != nil {
+		return specerrors.ValidationErrors{
+			specerrors.NewStructuredError(
+				err,
+				specerrors.UnassignedCode,
+			),
+		}
+	}
+	return nil
+}
+
+func shouldValidateDocsStructure(fsys fspath.FS) (*specerrors.DocsStructureEnforced, error) {
+	validationPath := "validation.yml"
+	files, err := pkgpath.Files(fsys, validationPath)
+	if err != nil || len(files) == 0 {
+		return nil, nil
+	}
+
+	validationFile := files[0]
+	data, err := validationFile.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file %s: %w", validationFile.Path(), err)
+	}
+
+	var config specerrors.ConfigFilter
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal YAML from %s: %w", validationFile.Path(), err)
+	}
+
+	if config.DocsStructureEnforced.Enabled {
+		return &config.DocsStructureEnforced, nil
+	}
+
+	return nil, nil
+}
+
+func readDocsStructureEnforcedConfig(fsys fspath.FS, config *specerrors.DocsStructureEnforced) ([]string, error) {
+	defaultSections, err := loadSectionsFromConfig(fmt.Sprintf("%d", config.Version))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load enforced sections from config: %w", err)
+	}
+
+	var sections []string
+	for _, section := range defaultSections {
+		if contains(config.DocsStructureSkip, section) {
+			continue
+		}
+		sections = append(sections, section)
+	}
+
+	return sections, nil
+}
+
+func contains(slice []specerrors.DocsStructureSkip, item string) bool {
+	for _, s := range slice {
+		if s.Title == item {
+			return true
+		}
+	}
+	return false
+}
+
+func loadSectionsFromConfig(version string) ([]string, error) {
+	var schemaPath string
+	switch version {
+	case "1":
+		schemaPath = "enforced_sections_v1.yml"
+	default:
+		return nil, fmt.Errorf("unsupported version: %s", version)
+	}
+
+	data, err := fs.ReadFile(spec.DocsFS(), schemaPath)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to read schema file: %w", err)
+	}
+
+	var spec EnforcedSections
+	if err := yaml.Unmarshal(data, &spec); err != nil {
+		return nil, fmt.Errorf("invalid schema YAML: %w", err)
+	}
+
+	if spec.Version != version {
```

Comment:
How can this happen? Would it be an spec with an incorrect version?

I think we could rely on the name of the file, so we don't need the redundant `version` inside the spec.
```
schemaPath := fmt.Sprintf("enforced_sections_v%s.yml", version)
```